src/util/modularUtil0.ml

```diff
@@ -0,0 +1,12 @@
+open GoblintCil
+open GobConfig
+let type_to_varinfo t =
+  TypeVarinfoMap.to_varinfo t
+
+let varinfo_to_canonical x =
+  let t = x.vtype in
+  type_to_varinfo t
+
+(** If [x] is global and we are in ["modular"] mode, retrieves the varinfo_to_canonical for [x], else returns [x] itself *)
+let varinfo_or_canonical x =
+  if get_bool "modular" && x.vglob then varinfo_to_canonical x else x
```

Comment:
This function probably gets called all the time around every `varinfo`, right? In that case it might later be worth caching the option value into a `bool ref` variable.